### ISSUE TITLE
Kernel_task memory leak

### ISSUE THREAD

**Issue by TheGodfatherDog** [checkra1n/BugTracker#2114]
**Tell us about your setup:**
1. iPhone 7
2. 14.4
3. 0.12.4
4. odysseyn1x and checkn`x iso images
5. USB-A

After jailbreak checkra1n or odysseyra1n, the amount of memory used by kernel_task is constantly increasing. At first, ldrestart helped to reset its value to 300 MB, but after a couple of days it does not help. The process takes all the memory and applications start to crash. Everything works fine without jailbreak. It doesn't depend on my list of tweaks. Even without tweaks (I checked) this problem happens over time. If to close all applications, the memory consumption of the process will decrease, but only slightly. iPhone 7, ios 14.4. Checkra1n and odysseyra1n last several versions. Safe mode does not solve the problem. To use the phone at least somehow, I need to restart it constantly.

Screenshot Cocoatop:
https://preview.redd.it/8bi1y0evo2171.png?width=750&format=png&auto=webp&s=33cc9f93799d5bf4e63c169debf433f5aea1fed4

**Comment by TheGodfatherDog** [comment]
This problem occurs with many people, it makes it impossible to use the iPhone in jailbreak mode.

**Comment by bitbribit** [comment]
it's a problem with A10 devices (iPad 6th gen and iPhone 7).

**Comment by theguywiththeproblem** [comment]
Yep. Have the same issue here. I messaged with siguza and after a couple days trying to troubleshoot. Nothing. Nothing seems wrong yet the issue clearly persists. There has to be something all these users are doing that is causing this issue.

**Comment by My-ThrowAwayAccount** [comment]
> 
> 
> it's a problem with A10 devices (iPad 6th gen and iPhone 7).

I'm having the same issue with an iPhone X (A11)

**Comment by theguywiththeproblem** [comment]
> I'm having the same issue with an iPhone X (A11)

Do apps begin to get killed more after about 3 days to the point where only 1 app can be opened at any time. 

If so, try this. Restore system in checkra1n. Then rejailbreak and do not install a bootstrap. No cydia or odysseyra1n. Then use your device like normal. 

If the issue reappears, then congratulations. apparently a11 is affected. If not then you’re fine.

One last thing, if the issue persists without a bootstrap, try restoring the system again in checkra1n, and resetting settings in settings. I’ve heard that this may help.

**Comment by My-ThrowAwayAccount** [comment]
> Do apps begin to get killed more after about 3 days to the point where only 1 app can be opened at any time.

No, the issues I'm having are different; kernel_task uses 20-23% of my RAM but it can still bare several applications at the same time, the issue is most prominent with my keyboard that crashes every 3 seconds, I can't type a sentence without it dying and, in fact, I don't even need to: it dies even if I don't type on it, as long as it's on-screen it's guaranteed to restart.
The phone does randomly slow down for a few minutes every once in a while, it becomes so slow I can barely type one word in 5 seconds but crashing only happened twice in many months I've been jailbroken and the phone had been on for one week or more so I assumed that was the reason (little did I know kernel_task was slowly but surely eating my RAM more than Chrome on my laptop)
I'm having another issue with jailbreak at the moment and I need to completely remove and re-jailbreak the phone so I might try to not install Cydia and see if RAM usage is still that high

**Comment by My-ThrowAwayAccount** [comment]
> If the issue reappears, then congratulations. apparently a11 is affected. If not then you’re fine.

I just finished restoring (after one of the scariest experiences I've had with checkra1n, I was stuck in DFU, force restarting would just bring me back to DFU; luckily a software similar to the one pointed in [this GitHub comment](https://github.com/checkra1n/BugTracker/issues/462#issuecomment-607635760) saved my life and brought it back without updates or anything) and opened "Usage", which doesn't tell me which processes use my RAM but it tells me the usage. It's the same as before, I only have 50-150mb of free RAM, but what's interesting is that it's the same even when the phone is not jailbroken (I've not installed any bootstrap, Cydia is not installed and yet rebooting the phone only helps it by a mere 9%, going from 88% usage when jailbroken to 79%), so I might need to investigate this more

**Comment by theguywiththeproblem** [comment]
> I only have 50-150mb of free RAM, but what's interesting is that it's the same even when the phone is not jailbroken

From what I understand, the total ram usage is the same jailbroken or not. Keep using your device the way it is. If the issue reappears go to the hack different server (https://discord.gg/NAxRYvysuc) and ask about it in the checkra1n channel. They’ll probably help you. You could even ask siguza if you wanted. Hope you can get to the bottom of this.

**Comment by My-ThrowAwayAccount** [comment]
So you're saying that kernel_task uses 20% of RAM on never jailbroken devices as well? That's kind of a big memory leak...

**Comment by theguywiththeproblem** [comment]
> So you're saying that kernel_task uses 20% of RAM on never jailbroken devices as well? That's kind of a big memory leak...

Sorry for not making myself clear. What I meant to was that the amount of free ram in total isn’t really affected by being jailbroken or not so there’s no need to worry about free ram.

**Comment by My-ThrowAwayAccount** [comment]
I see, I was expecting to have more free ram, since the bug wouldn't be using all of it... I'm definitely still having issues, though. My keyboard keeps crashing 'cause it doesn't have enough RAM to operate (I'm using Microsoft SwiftKey). I'll try to reboot and see if the issue is still there without jailbreak. If that's the case I think my only option will be to backup my phone and completely wipe it - I hope that's not the case, though

**Comment by theguywiththeproblem** [comment]
> My keyboard keeps crashing 'cause it doesn't have enough RAM to operate (I'm using Microsoft SwiftKey). 

Try using the stock keyboard instead. Have to narrow down what’s causing this.

**Comment by My-ThrowAwayAccount** [comment]
The stock keyboard only crashed on me once, even when the SwiftKey keyboard constantly crashes. I've tried to allocate more RAM to third party keyboards (following a tutorial on reddit, giving it 128mb instead of 62 editing a plist file), but nothing really changed. It still crashes the same way and just as often (which is every 2-5 seconds). I have tried uninstalling and re-installing SwiftKey and even joining their TestFlight program in the past but it didn't really fix the issue.
Removing the custom theme from SwiftKey (which is not a tweak, SwiftKey lets you use a photo as background for your keyboard), so using one of the stock themes, did reduce the crashing, but it indeed still crashed. I feel we might be going out of topic here, but I've never really had the keyboard crash on me before the first time I've jailbroken my phone

**Comment by theguywiththeproblem** [comment]
> The stock keyboard only crashed on me once, even when the SwiftKey keyboard constantly crashes. 

Okay so use that for the time being. If you still have the issue with no bootstrap. You should probably go to the discord server I sent the link for. The folks there know a lot more than I and you’ll get help a lot faster than waiting for a response on an issue.

**Comment by theguywiththeproblem** [comment]
@My-ThrowAwayAccount Any updates on your issues?

**Comment by ZDSDDDD** [comment]
> This problem occurs with many people, it makes it impossible to use the iPhone in jailbreak mode.

Me too, 3 or 5days must be restart the iphone
